04/main.go

```diff
@@ -37,16 +37,12 @@ func main() {
 type app struct {
 	weaver.Implements[weaver.Main]
 	searcher weaver.Ref[Searcher]
+	lis      weaver.Listener `weaver:"emojis"`
```

Comment:
Do you find it weird to name this field `emojis`? I find it somewhat weird too. Does this mean our API isn't great?

Reply:
I found it weird to name this `emojis` because there's already a global `emojis` variable of type `map[string][]string` that maps emojis to their tags. I like the listener API! I opened a PR on the `weaver` repo to explain `weaver:"name"` tags earlier so people are even more aware of them.